README.md

```diff
@@ -1291,6 +1292,186 @@ let release () =
   decr in_use
 ```
 
+### Conditions
+
+[Eio.Condition][] allows a fiber to wait until some condition is true.
+For example:
+
+```ocaml
+module X = struct
+  (* Note: this version is not safe to share across domains! *)
+
+  let x = ref 0
+  let changed = Eio.Condition.create ()
+
+  let await_zero () =
+    while !x <> 0 do Eio.Condition.await_no_mutex changed done;
+    traceln "x is now zero"
+
+  let set v =
+    x := v;
+    Eio.Condition.broadcast changed;
+    traceln "x set to %d" v
+end
+```
+
+```ocaml
+# Eio_mock.Backend.run @@ fun () ->
+  X.set 5;
+  Fiber.both
+    (fun () ->
+       traceln "Waiting for x to be 0";
+       X.await_zero ()
+    )
+    (fun () -> X.set 0);;
++x set to 5
++Waiting for x to be 0
++x set to 0
++x is now zero
+- : unit = ()
+```
+
+Note that we need a loop in `await_zero`.
+This is needed because it's possible that another fiber might set it to zero
+and then set it to something else before the waiting fiber resumes.
+
+The above version is not safe to share across domains, because `await_zero` relies on the value of `x` not changing
+after it reads it but before `await_no_mutex` registers itself with the condition.
+Here's a domain-safe version:
+
+```ocaml
+module Y = struct
+  (* Safe to share between domains. *)
+
+  let y = ref 0
+  let mutex = Eio.Mutex.create ()
+  let changed = Eio.Condition.create ()
+
+  let await_zero () =
+    Eio.Mutex.use_ro mutex (fun () ->
+      while !y <> 0 do Eio.Condition.await changed mutex done;
+      traceln "y is now zero (at least until we release the mutex)"
+    )
+
+  let set v =
+    Eio.Mutex.use_rw mutex ~protect:true (fun () ->
+       y := v;
+       Eio.Condition.broadcast changed;
+       traceln "y set to %d" v
+    );
+end
+```
+
+Here, `Eio.Condition.await` registers itself with `changed` and only then releases the mutex,
+allowing other threads to change `y`. When it gets woken, it re-acquires the mutex.
+
+```ocaml
+# Eio_mock.Backend.run @@ fun () ->
+  Y.set 5;
+  Fiber.both
+    (fun () ->
+       traceln "Waiting for y to be 0";
+       Y.await_zero ()
+    )
+    (fun () -> Y.set 0);;
++y set to 5
++Waiting for y to be 0
++y set to 0
++y is now zero (at least until we release the mutex)
+- : unit = ()
+```
+
+Conditions are more difficult to use correctly than e.g. promises or streams.
+In particular, it is easy to miss a notification due to `broadcast` getting called before `await`.
+However, they can be useful if used carefully.
+
+### Example: Signal handlers
+
+On Unix-type systems, processes can react to *signals*.
```

Comment:
I think this holds for windows as well, no clue to what extend though.

Reply:
I don't think so; if you try to ignore SIGPIPE on Windows then it raises an exception, for example (see #420).